File: modules/role_manager.py
import discord
from discord.ext import commands

class RoleManagerCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot
        self.allowed_user_id = 818166724641030193  # Replace with your Discord user ID

    @commands.command()
    async def addrole(self, ctx, member: discord.Member, *, role: discord.Role):
        """Adds a role to a member."""
        if ctx.author.id != self.allowed_user_id:
            await ctx.send("You are not authorized to use this command.")
            return

        if role in member.roles:
            await ctx.send(f"{member.display_name} already has the role {role.name}.")
            return

        try:
            await member.add_roles(role)
            await ctx.send(f"Successfully added role {role.name} to {member.display_name}.")
        except discord.Forbidden:
            await ctx.send("I do not have permission to add this role.")
        except Exception as e:
            await ctx.send(f"An error occurred: {e}")

    @commands.command()
    async def removerole(self, ctx, member: discord.Member, *, role: discord.Role):
        """Removes a role from a member."""
        if ctx.author.id != self.allowed_user_id:
            await ctx.send("You are not authorized to use this command.")
            return

        if role not in member.roles:
            await ctx.send(f"{member.display_name} does not have the role {role.name}.")
            return

        try:
            await member.remove_roles(role)
            await ctx.send(f"Successfully removed role {role.name} from {member.display_name}.")
        except discord.Forbidden:
            await ctx.send("I do not have permission to remove this role.")
        except Exception as e:
            await ctx.send(f"An error occurred: {e}")

    @commands.command()
    async def listroles(self, ctx, member: discord.Member):
        """Lists all roles of a member."""
        if ctx.author.id != self.allowed_user_id:
            await ctx.send("You are not authorized to use this command.")
            return

        roles = [role.name for role in member.roles if role.name != "@everyone"]
        if roles:
            await ctx.send(f"{member.display_name} has the following roles: {', '.join(roles)}")
        else:
            await ctx.send(f"{member.display_name} has no roles.")

async def setup(bot):
    await bot.add_cog(RoleManagerCog(bot))

File: modules/role_manager.js
const { SlashCommandBuilder, MessageFlags } = require('discord.js');

module.exports = {
  data: new SlashCommandBuilder()
    .setName('role')
    .setDescription('Manage roles')
    .addSubcommand(sc => sc
      .setName('add')
      .setDescription('Add a role to a member')
      .addUserOption(o => o.setName('member').setDescription('Member').setRequired(true))
      .addRoleOption(o => o.setName('role').setDescription('Role').setRequired(true)))
    .addSubcommand(sc => sc
      .setName('remove')
      .setDescription('Remove a role from a member')
      .addUserOption(o => o.setName('member').setDescription('Member').setRequired(true))
      .addRoleOption(o => o.setName('role').setDescription('Role').setRequired(true)))
    .addSubcommand(sc => sc
      .setName('list')
      .setDescription('List roles of a member')
      .addUserOption(o => o.setName('member').setDescription('Member').setRequired(true))),
  async execute(interaction) {
    if (!interaction.client.isAdmin(interaction.member || interaction.user)) {
      await interaction.reply({ content: 'Unauthorized.', flags: MessageFlags.Ephemeral });
      return;
    }
    const sub = interaction.options.getSubcommand();
    const member = interaction.options.getMember('member');
    const role = interaction.options.getRole('role');
    try {
      if (sub === 'add') {
        if (member.roles.cache.has(role.id)) {
          await interaction.reply(`${member.displayName} already has the role ${role.name}.`);
        } else {
          await member.roles.add(role);
          await interaction.reply(`Successfully added role ${role.name} to ${member.displayName}.`);
        }
      } else if (sub === 'remove') {
        if (!member.roles.cache.has(role.id)) {
          await interaction.reply(`${member.displayName} does not have the role ${role.name}.`);
        } else {
          await member.roles.remove(role);
          await interaction.reply(`Successfully removed role ${role.name} from ${member.displayName}.`);
        }
      } else if (sub === 'list') {
        const roles = member.roles.cache.filter(r => r.name !== '@everyone').map(r => r.name);
        await interaction.reply(roles.length ? `${member.displayName} has the following roles: ${roles.join(', ')}` : `${member.displayName} has no roles.`);
      }
    } catch (e) {
      await interaction.reply(`An error occurred: ${e.message || e}`);
    }
  }
};
